assets/src/pages/room/components/MediaControlButtons.tsx

```diff
@@ -65,36 +54,72 @@ const getAutomaticControls = (
           cameraStreaming.setActive(true);
         },
       },
-  displayMedia.stream
+  userMediaAudio.isEnabled
     ? {
-        id: "stream-stop",
-        icon: "/svg/computer-line.svg",
-        hover: "Stop the screensharing",
+        id: "mic-mute",
+        icon: Microphone,
+        hover: "Turn off the microphone",
+        className: neutralButtonStyle,
         onClick: () => {
-          displayMedia.stop();
-          screenSharingStreaming.setActive(false);
+          userMediaAudio.disable();
+          audioStreaming.setActive(false);
         },
       }
     : {
-        id: "stream-start",
-        icon: "/svg/computer-line.svg",
-        hover: "Start the screensharing",
+        id: "mic-unmute",
+        icon: MicrophoneOff,
+        hover: "Turn on the microphone",
+        className: activeButtonStyle,
         onClick: () => {
-          displayMedia.start();
-          screenSharingStreaming.setActive(true);
+          if (userMediaAudio.stream) {
+            userMediaAudio.enable();
+          } else {
+            userMediaAudio.start();
+          }
+          audioStreaming.setActive(true);
         },
       },
   {
     id: "leave-room",
-    icon: "/svg/phone-fill.svg",
+    icon: HangUp,
     hover: "Leave the room",
-    imgClasses: "black-to-red transform rotate-135",
+    className: redButtonStyle,
     onClick: () => {
       navigate("/");
     },
   },
+  displayMedia.stream
+    ? {
+        id: "stream-stop",
+        icon: Screenshare,
+        hover: "Stop sharing your screen",
+        className: neutralButtonStyle,
+        onClick: () => {
+          displayMedia.stop();
+          screenSharingStreaming.setActive(false);
+        },
+      }
+    : {
+        id: "stream-start",
+        icon: Screenshare,
+        hover: "Share your screen",
+        className: neutralButtonStyle,
+        onClick: () => {
+          displayMedia.start();
+          screenSharingStreaming.setActive(true);
+        },
+      },
+  //TODO enable when chat is implemented
+  // {
+  //   id: "chat",
+  //   icon: Chat,
+  //   hover: "Open the chat",
+  //   className: neutralButtonStyle,
+  //   onClick: () => undefined,
+  // },
 ];
 
+//dev helpers
 const getManualControls = (
```

Comment:
This function mainly consists of a nested structure made out of 150 lines of tertiary operators and java script objects. Have you considered extracting some of the icons in a form of a component (like `MicrophoneIcon` etc) with the data from here given as prop values?

Similar thing was done in (MembraneLive)[https://github.com/membraneframework-labs/membrane_live/blob/master/assets/js/components/event/ControlPanel.tsx]

Reply:
I agree with you that it might be worth refactoring this logic a little bit. I’ll create a separate task for this.

Reply:
OK, please link the issue here

Reply:
https://membraneframework.atlassian.net/browse/MV-260